atom/common/promise_util.h

```diff
@@ -72,6 +82,16 @@ class Promise : public base::RefCounted<Promise> {
                               mate::ConvertToV8(isolate(), value));
   }
 
+  template <typename T>
+  v8::MaybeLocal<v8::Promise> Then(const T& value) {
```

Comment:
IMO this should not be a `template`'ed function and instead explicitly take a `base::Callback` and do the `mate::ConvertToV8` call that we need to do 🤔